src/registration/validators.ts

```diff
@@ -0,0 +1,25 @@
+import { loadCSR } from '@zbayapp/identity'
+import { registerDecorator, ValidationOptions, ValidationArguments } from 'class-validator'
+
+export function IsCsr(validationOptions?: ValidationOptions) {
+  return function (object: Object, propertyName: string) {
+    registerDecorator({
+      name: 'isCsr',
+      target: object.constructor,
+      propertyName: propertyName,
+      options: validationOptions,
+      validator: {
+        async validate(value: any, _args: ValidationArguments) {
+          const prom: Promise<boolean> = new Promise(resolve => {
+            loadCSR(value).then(() => {
```

Comment:
Just occured to me, there is probably a method on CSR that checks that its valid ( that the signature included is to publicmkey and parameters its claiming). It would be good to call it here

Reply:
Did this get resolved?

Reply:
Csr verification is being done outside this validator (in Storage.saveCertificate).